app/controllers/auth/sessions_controller.rb

```diff
@@ -184,6 +189,12 @@ def on_authentication_failure(user, security_measure, failure_reason)
     UserMailer.failed_2fa(user, request.remote_ip, request.user_agent, Time.now.utc).deliver_later!
   end
 
+  def log_failed_authentication(user)
+    ip_address = request.remote_ip
+    auth_logger = Logger.new(Rails.root.join('log', 'authentication.log'))
+    auth_logger.warn("Failed login attempt for user: #{user || 'Unknown'}, IP: #{ip_address}")
+  end
```

Comment:
I would not create a new file-based logger, especially since logs would be unusual in this place, might not be able to be written to (causing errors), or might stay up for an arbitrary long amount of time without usual cleanup or log rotation.